From 5fd63363eb29c7e4d3d46762606d6bebe880f179 Mon Sep 17 00:00:00 2001
From: David Grayson <davidegrayson@gmail.com>
Date: Mon, 16 Feb 2026 08:40:24 -0800
Subject: [PATCH] headers/esent: Fix JetOpenTemporaryTable compilation error.

Only declare JetOpenTemporaryTable if we also define its
second argument type, JET_OPENTEMPORARYTABLE.

Signed-off-by: David Grayson <davidegrayson@gmail.com>
---
 mingw-w64-headers/include/esent.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/mingw-w64-headers/include/esent.h b/mingw-w64-headers/include/esent.h
index b3272285a..2cd3bb729 100644
--- a/mingw-w64-headers/include/esent.h
+++ b/mingw-w64-headers/include/esent.h
@@ -2559,10 +2559,12 @@ JET_ERR JET_API JetOpenTableW(
 
 #define JetOpenTable __MINGW_NAME_AW(JetOpenTable)
 
+#if (JET_VERSION >= 0x0600)
 JET_ERR JET_API JetOpenTemporaryTable(
   JET_SESID sesid,
   JET_OPENTEMPORARYTABLE* popentemporarytable
 );
+#endif /*(JET_VERSION >= 0x0600)*/
 
 JET_ERR JET_API JetOpenTempTable(
   JET_SESID sesid,
-- 
2.43.0

